node/src/main/scala/org/bitcoins/node/NodeType.scala

```diff
@@ -6,21 +6,29 @@ sealed abstract class NodeType {
   def shortName: String
 }
 
+sealed abstract class InternalImplementationNodeType extends NodeType
+sealed abstract class ExternalImplementationNodeType extends NodeType
+
 object NodeType extends StringFactory[NodeType] {
 
-  final case object FullNode extends NodeType {
+  final case object FullNode extends InternalImplementationNodeType {
     override def shortName: String = "full"
   }
 
-  final case object NeutrinoNode extends NodeType {
+  final case object NeutrinoNode extends InternalImplementationNodeType {
     override def shortName: String = "neutrino"
   }
 
-  final case object SpvNode extends NodeType {
+  final case object SpvNode extends InternalImplementationNodeType {
     override def shortName: String = "spv"
   }
 
-  val all: Vector[NodeType] = Vector(FullNode, NeutrinoNode, SpvNode)
+  final case object BitcoindBackend extends ExternalImplementationNodeType {
```

Comment:
Hmm, i'm not sure if I agree with this being considered a `NodeType`. It's more of a source for chain information. Would you consider the block stream API another node type as well if we were to implement that?

I guess `BitcoindRpcClient` does implement `NodeApi` though so :man_shrugging:.

Reply:
I agree this is kinda bastardizing the `NodeType` but probably a better solution can be done in the future.